File: app/src/main/java/co/shek/kkiapayandroidsample/MainActivity.java
package co.shek.kkiapayandroidsample;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import co.opensi.kkiapay.STATUS;
import co.opensi.kkiapay.uikit.Kkiapay;
import kotlin.Unit;
import kotlin.jvm.functions.Function2;


public class MainActivity extends AppCompatActivity {
    @Override
    public void onCreate (@Nullable Bundle savedInstanceState) {
        super.onCreate (savedInstanceState);
        setContentView(R.layout.activity_main);

        Button testButtonWithKkiapay = (Button) findViewById(R.id.test_button_with_kkiapay);
        testButtonWithKkiapay.setOnClickListener (new View.OnClickListener () {
            @Override
            public void onClick (View view) {
                Kkiapay.get().requestPayment(MainActivity.this,
                        "1",
                        "Paiement de services",
                        "Nom Prenom", "");
            }
        });

    }

    @Override
    protected void onStart () {
        super.onStart ();
        // Setup listener for payment status
        Kkiapay.get().setListener(new Function2<STATUS, String, Unit> () {
            @Override
            public Unit invoke(STATUS status, String s) {
                //The following code will be run when user will end the payment
                Toast.makeText(MainActivity.this, "Transaction: ${status.name} -> $transactionId", Toast.LENGTH_LONG).show();
                return null;
            }
        });
    }

    @Override
    protected void onActivityResult (int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult (requestCode, resultCode, data);
        Kkiapay.get().handleActivityResult(requestCode, resultCode, data);
    }
}

File: app/src/main/java/co/shek/kkiapayandroidsample/MainActivity.kt
package co.shek.kkiapayandroidsample

import co.opensi.kkiapay.uikit.Kkiapay.get
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import co.opensi.kkiapay.STATUS
import android.widget.Toast
import android.content.Intent
import android.view.View
import android.widget.Button
import co.opensi.kkiapay.uikit.Kkiapay

class MainActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        super.onStart()

        // Setup listener for payment status
        Kkiapay.get().setListener { status, transactionId ->

            //The following code will be run when user will end the payment
            Toast.makeText(this@MainActivity, "Transaction: ${status.name} -> $transactionId", Toast.LENGTH_LONG).show()
        }
    }

    override fun onResume() {
        super.onResume()
        val testButtonWithKkiapay = findViewById<View>(R.id.test_button_with_kkiapay) as Button
        testButtonWithKkiapay.setOnClickListener {
            Kkiapay.get().requestPayment(this, "1",
                reason = "Payment of awesome service", name = "Johna DOE",
                sandbox = true, //countries = listOf("CI"), partnerId = "AxXxxXXid",
                phone = "2250577100000", //email = "email@mail.com", paymentMethods = listOf("momo","card","direct_debit")
            )
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        Kkiapay.get().handleActivityResult(requestCode, resultCode, data)
    }
}